tsl/test/expected/chunk_column_stats.out

```diff
@@ -0,0 +1,282 @@
+       time TIMESTAMP WITH TIME ZONE NOT NULL,
+       sensor_id INTEGER NOT NULL,
+       cpu double precision null,
+       temperature double precision null,
+       name varchar(100) default 'this is a default string value'
+);
+CREATE INDEX sense_idx ON sample_table (sensor_id);
+SELECT * FROM create_hypertable('sample_table', 'time',
+       chunk_time_interval => INTERVAL '2 months');
+WARNING:  column type "character varying" used for "name" does not follow best practices
+ hypertable_id | schema_name |  table_name  | created 
+---------------+-------------+--------------+---------
+             1 | public      | sample_table | t
+(1 row)
+
+\set start_date '2022-01-28 01:09:53.583252+05:30'
+INSERT INTO sample_table
+    SELECT
+       	time + (INTERVAL '1 minute' * random()) AS time,
+       		sensor_id,
+       		random() AS cpu,
+       		random()* 100 AS temperature
+       	FROM
+       		generate_series(:'start_date'::timestamptz - INTERVAL '1 months',
+                            :'start_date'::timestamptz - INTERVAL '1 week',
+                            INTERVAL '1 hour') AS g1(time),
+       		generate_series(1, 8, 1 ) AS g2(sensor_id)
+       	ORDER BY
+       		time;
+\set start_date '2023-03-17 17:51:11.322998+05:30'
+-- insert into new chunks
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 12, 21.98, 33.123, 'new row1');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 12, 17.66, 13.875, 'new row1');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 13, 21.98, 33.123, 'new row2');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 9, 21.98, 33.123, 'new row2');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 14, 21.98, 33.123, 'new row2');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 15, 0.988, 33.123, 'new row3');
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 16, 4.6554, 47, 'new row3');
+-- Non-int, date, timestamp cannot be specified as a min/max ranges for now
+-- We could expand to FLOATs, NUMERICs later
+\set ON_ERROR_STOP 0
+SELECT * FROM enable_column_stats('sample_table', 'name');
+ERROR:  data type "character varying" unsupported for range calculation
+\set ON_ERROR_STOP 1
+-- Specify tracking of min/max ranges for a column
+SELECT * FROM enable_column_stats('sample_table', 'sensor_id');
+ column_stats_id | created 
+-----------------+---------
+               1 | t
+(1 row)
+
+-- The above should add an entry with MIN/MAX int64 entries for invalid chunk id
+-- to indicate that ranges on this column should be calculated for chunks
+SELECT * from _timescaledb_catalog.chunk_column_stats;
+ id | hypertable_id | chunk_id | column_name | column_type |     range_start      |      range_end      | valid 
+----+---------------+----------+-------------+-------------+----------------------+---------------------+-------
+  1 |             1 |        0 | sensor_id   | integer     | -9223372036854775808 | 9223372036854775807 | t
```

Comment:
@erimatnor 

If DML happens into a
    compressed chunk then as part of marking it as partial, we also mark
    the corresponding catalog entries as "invalid". So partial chunks do
    not get excluded further. When recompression happens we get the new
    min/max ranges from the uncompressed portion and try to reconcile the
    ranges in the catalog based on these new values. This is safe to do in
    case of INSERTs and UPDATEs. In case of DELETEs, since we are deleting
    rows, it's possible that the min/max ranges change, but as of now we
    err on the side of caution and retain the earlier values which can be
    larger than the actual range.